pkg/registry/generic/etcd/etcd.go

```diff
@@ -409,10 +409,18 @@ func (e *Etcd) Delete(ctx api.Context, name string, options *api.DeleteOptions)
 	if options.Preconditions != nil {
 		preconditions.UID = options.Preconditions.UID
 	}
-	graceful, pendingGraceful, err := rest.BeforeDelete(e.DeleteStrategy, ctx, obj, options)
-	if err != nil {
-		return nil, err
+
+	// DeleteStrategy is doc'ed as optional, but without one you can't be graceful or you'll panic
+	// tolerate an optional field being optional
```

Comment:
I think you mean "Assume non-graceful deletes if no delete strategy is set"? I can't parse this comment as it is currently.